tests/test_loadDiaCatalogs.py

```diff
@@ -0,0 +1,357 @@
+            dir=os.path.dirname(__file__))
+
+        self.apdbConfig = ApdbConfig()
+        self.apdbConfig.db_url = "sqlite:///" + self.db_file
+        self.apdbConfig.isolation_level = "READ_UNCOMMITTED"
+        self.apdbConfig.dia_object_index = "baseline"
+        self.apdbConfig.dia_object_columns = []
+        self.apdbConfig.schema_file = _data_file_name(
+            "apdb-schema.yaml", "dax_apdb")
+        self.apdbConfig.column_map = _data_file_name(
+            "apdb-ap-pipe-afw-map.yaml", "ap_association")
+        self.apdbConfig.extra_schema_file = _data_file_name(
+            "apdb-ap-pipe-schema-extra.yaml", "ap_association")
+
+        self.apdb = Apdb(config=self.apdbConfig,
+                         afw_schemas=dict(DiaObject=make_dia_object_schema(),
+                                          DiaSource=make_dia_source_schema()))
+        self.apdb.makeSchema()
+
+        # Expected HTM pixel ranges for max range=4 and level = 20. This
+        # set of pixels should be same for the WCS created by default in
+        # makeExposure and for one with a flipped y axis.
+        self.ranges = np.sort(
+            np.array([(15154776375296, 15154779521024),
+                      (15154788958208, 15154792103936)]).flatten())
+
+        self.pixelator = sphgeom.HtmPixelization(20)
+        self.exposure = makeExposure(False, False)
+
+        self.diaObjects = makeDiaObjects(20, self.exposure, self.pixelator)
+        self.diaSources = makeDiaSources(
+            100,
+            self.diaObjects["diaObjectId"].to_numpy(),
+            self.exposure,
+            self.pixelator)
+
+        self.dateTime = dafBase.DateTime("2014-05-13T17:00:00.000000000",
+                                         dafBase.DateTime.Timescale.TAI)
+
+        self.apdb.storeDiaSources(self.diaSources)
+        self.apdb.storeDiaObjects(self.diaObjects, self.dateTime.toPython())
+
+    def tearDown(self):
+        del self.tmp_file
+        os.remove(self.db_file)
+        del self.db_file
+
+    def testRun(self):
+        """Test the full run method for the loader.
+        """
+        diaLoader = LoadDiaCatalogsTask()
+        result = diaLoader.run(self.exposure, self.apdb)
+
+        self.assertEqual(len(result.diaObjects), len(self.diaObjects))
+        self.assertEqual(len(result.diaSources), len(self.diaSources))
+
+    def testLoadDiaObjects(self):
+        """Test that the correct number of diaObjects are loaded.
+        """
+        diaLoader = LoadDiaCatalogsTask()
+        normPixels = diaLoader._getPixelRanges(self.exposure)
+        diaObjects = diaLoader.loadDiaObjects(normPixels,
+                                              self.apdb)
+        self.assertEqual(len(diaObjects), len(self.diaObjects))
+
+    def testLoadDiaSources(self):
+        """Test that the correct number of diaSources are loaded.
+
+        Also check that they can be properly loaded both by location and
+        ``diaObjectId``.
```

Comment:
It might be better to make these two separate tests, just to make sure there's no cross-talk.